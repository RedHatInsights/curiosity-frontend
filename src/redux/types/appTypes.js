/**
 * @memberof Types
 * @module AppTypes
 */

const STATUS_4XX = '4XX';
const STATUS_5XX = '5XX';
const SET_PRODUCT = 'SET_PRODUCT';
const SET_PRODUCT_VARIANT = 'SET_PRODUCT_VARIANT';
const SET_PRODUCT_VARIANT_QUERY_RESET_ALL = 'SET_PRODUCT_VARIANT_QUERY_RESET_ALL';

/**
 * Application action, reducer types.
 *
 * @type {{STATUS_4XX: string, SET_PRODUCT_VARIANT: string, SET_PRODUCT_VARIANT_QUERY_RESET_ALL: string,
 *     SET_PRODUCT: string, STATUS_5XX: string}}
 */
const appTypes = {
  STATUS_4XX,
  STATUS_5XX,
  SET_PRODUCT,
  SET_PRODUCT_VARIANT,
  SET_PRODUCT_VARIANT_QUERY_RESET_ALL
};

export {
  appTypes as default,
  appTypes,
  STATUS_4XX,
  STATUS_5XX,
  SET_PRODUCT,
  SET_PRODUCT_VARIANT,
  SET_PRODUCT_VARIANT_QUERY_RESET_ALL
};
