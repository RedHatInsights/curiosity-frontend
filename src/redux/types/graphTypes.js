/**
 * @memberof Types
 * @module GraphTypes
 */

const SET_GRAPH_LEGEND = 'SET_GRAPH_LEGEND';

/**
 * Graph action, reducer types.
 *
 * @type {{SET_GRAPH_LEGEND: string}}
 */
const graphTypes = {
  SET_GRAPH_LEGEND
};

export { graphTypes as default, graphTypes, SET_GRAPH_LEGEND };
