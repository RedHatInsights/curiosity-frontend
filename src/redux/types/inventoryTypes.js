const SET_INVENTORY_TAB = 'SET_INVENTORY_TAB';

/**
 * Inventory action, reducer types.
 *
 * @type {{SET_INVENTORY_TAB: string}}
 */
const inventoryTypes = {
  SET_INVENTORY_TAB
};

export { inventoryTypes as default, inventoryTypes, SET_INVENTORY_TAB };
