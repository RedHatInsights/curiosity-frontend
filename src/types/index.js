import { rhsmApiTypes } from './rhsmApiTypes';

const apiTypes = {
  rhsmApi: rhsmApiTypes
};

export { apiTypes as default, apiTypes, rhsmApiTypes };
