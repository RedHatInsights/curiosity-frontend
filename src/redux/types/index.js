import { appTypes } from './appTypes';
import { graphTypes } from './graphTypes';
import { platformTypes } from './platformTypes';
import { queryTypes } from './queryTypes';
import { rhsmTypes } from './rhsmTypes';
import { toolbarTypes } from './toolbarTypes';
import { userTypes } from './userTypes';

const reduxTypes = {
  app: appTypes,
  graph: graphTypes,
  platform: platformTypes,
  query: queryTypes,
  rhsm: rhsmTypes,
  toolbar: toolbarTypes,
  user: userTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  appTypes,
  graphTypes,
  platformTypes,
  queryTypes,
  rhsmTypes,
  toolbarTypes,
  userTypes
};
