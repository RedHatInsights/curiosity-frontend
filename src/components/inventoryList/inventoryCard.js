import React from 'react';
import PropTypes from 'prop-types';
import _isEqual from 'lodash/isEqual';
import { SortByDirection, TableVariant } from '@patternfly/react-table';
import { Bullseye, Card, CardActions, CardBody, CardFooter, CardHeader, CardHeaderMain } from '@patternfly/react-core';
import { TableToolbar } from '@redhat-cloud-services/frontend-components/TableToolbar';
import _camelCase from 'lodash/camelCase';
import { helpers } from '../../common';
import { connect, reduxActions, reduxSelectors, reduxTypes, store } from '../../redux';
import Table from '../table/table';
import { Loader } from '../loader/loader';
import { MinHeight } from '../minHeight/minHeight';
import GuestsList from '../guestsList/guestsList';
import { inventoryCardHelpers } from './inventoryCardHelpers';
import Pagination from '../pagination/pagination';
import { ToolbarFieldDisplayName } from '../toolbar/toolbarFieldDisplayName';
import { paginationHelpers } from '../pagination/paginationHelpers';
import {
  RHSM_API_QUERY_INVENTORY_SORT_DIRECTION_TYPES as SORT_DIRECTION_TYPES,
  RHSM_API_QUERY_INVENTORY_SORT_TYPES as SORT_TYPES,
  RHSM_API_QUERY_SET_TYPES
} from '../../services/rhsm/rhsmConstants';
import { translate } from '../i18n/i18n';

/**
 * ToDo: refactor this component towards hooks.
 * This base component was copied from the original hosts component in order to speed
 * deliver the RHOSAK product views. This is an interim component and needs to be refactored.
 * Refactor will include
 * - Auth component, hook conversion.
 *    - Mock service updates for regular promise/platform calls
 *    - Mock service transform or hook for RBAC perms
 * - Auth context with "useAuthContext" for grabbing session object
 * - InventoryCard component
 * - Display name toolbar filter, and filter/query context
 */
/**
 * An instances interim inventory component.
 *
 * @augments React.Component
 * @fires onColumnSort
 * @fires onPage
 * @fires onUpdateInventoryData
 */
class InventoryList extends React.Component {
  componentDidMount() {
    this.onUpdateInventoryData();
  }

  componentDidUpdate(prevProps) {
    const { productId, query } = this.props;

    if (productId !== prevProps.productId || !_isEqual(query, prevProps.query)) {
      this.onUpdateInventoryData();
    }
  }

  /**
   * On column sort update state.
   *
   * @event onColumnSort
   * @param {object} data pass-through inventory data.
   * @param {object} sortParams
   * @param {string} sortParams.direction
   * @param {string} sortParams.id column identifier
   */
  onColumnSort = (data, { direction, id }) => {
    const { productId } = this.props;
    const updatedSortColumn = Object.values(SORT_TYPES).find(value => value === id || _camelCase(value) === id);
    let updatedDirection;

    if (!updatedSortColumn) {
      if (helpers.DEV_MODE || helpers.REVIEW_MODE) {
        console.warn(`Sorting can only be performed on select fields, confirm field ${id} is allowed.`);
      }
      return;
    }

    switch (direction) {
      case SortByDirection.desc:
        updatedDirection = SORT_DIRECTION_TYPES.DESCENDING;
        break;
      default:
        updatedDirection = SORT_DIRECTION_TYPES.ASCENDING;
        break;
    }

    store.dispatch([
      {
        type: reduxTypes.query.SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES[RHSM_API_QUERY_SET_TYPES.DIRECTION],
        viewId: productId,
        [RHSM_API_QUERY_SET_TYPES.DIRECTION]: updatedDirection
      },
      {
        type: reduxTypes.query.SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES[RHSM_API_QUERY_SET_TYPES.SORT],
        viewId: productId,
        [RHSM_API_QUERY_SET_TYPES.SORT]: updatedSortColumn
      }
    ]);
  };

  /**
   * On paging and on perPage events.
   *
   * @event onPage
   * @param {object} params
   * @param {number} params.offset
   * @param {number} params.perPage
   */
  onPage = ({ offset, perPage }) => {
    const { productId } = this.props;

    store.dispatch([
      {
        type: reduxTypes.query.SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES[RHSM_API_QUERY_SET_TYPES.OFFSET],
        viewId: productId,
        [RHSM_API_QUERY_SET_TYPES.OFFSET]: offset
      },
      {
        type: reduxTypes.query.SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES[RHSM_API_QUERY_SET_TYPES.LIMIT],
        viewId: productId,
        [RHSM_API_QUERY_SET_TYPES.LIMIT]: perPage
      }
    ]);
  };

  /**
   * Call the RHSM APIs, apply filters.
   *
   * @event onUpdateInventoryData
   */
  onUpdateInventoryData = () => {
    const { getInstancesInventory, isDisabled, productId, query } = this.props;

    if (!isDisabled && productId) {
      getInstancesInventory(productId, query);
    }
  };

  /**
   * Render an inventory table.
   *
   * @returns {Node}
   */
  renderTable() {
    const { filterGuestsData, filterInventoryData, data: listData, query, session, settings } = this.props;
    let updatedColumnHeaders = [];

    const updatedRows = listData.map(({ ...cellData }) => {
      const { columnHeaders, cells } = inventoryCardHelpers.parseRowCellsListData({
        filters: inventoryCardHelpers.parseInventoryFilters({
          filters: filterInventoryData,
          onSort: this.onColumnSort,
          query
        }),
        cellData,
        session
      });

      updatedColumnHeaders = columnHeaders;

      const guestsId = cellData?.subscriptionManagerId;
      let hasGuests = cellData?.numberOfGuests > 0 && guestsId;

      // Apply hasGuests callback, return boolean
      if (typeof settings?.hasGuests === 'function') {
        hasGuests = settings.hasGuests({ ...cellData }, { ...session });
      }

      return {
        cells,
        expandedContent:
          (hasGuests && (
            <GuestsList
              key={guestsId}
              filterGuestsData={filterGuestsData}
              numberOfGuests={cellData?.numberOfGuests}
              id={guestsId}
              query={query}
            />
          )) ||
          undefined
      };
    });

    return (
      <Table
        borders
        variant={TableVariant.compact}
        className="curiosity-inventory-list"
        columnHeaders={updatedColumnHeaders}
        rows={updatedRows}
      />
    );
  }

  /**
   * Render an inventory card.
   *
   * @returns {Node}
   */
  render() {
    const {
      error,
      filterInventoryData,
      fulfilled,
      isDisabled,
      data: listData,
      meta,
      pending,
      perPageDefault,
      query,
      t,
      viewId
    } = this.props;

    if (isDisabled) {
      return (
        <Card className="curiosity-inventory-card__disabled">
          <CardBody>
            <Bullseye>{t('curiosity-inventory.tab', { context: 'disabled' })}</Bullseye>
          </CardBody>
        </Card>
      );
    }

    const itemCount = meta?.count;
    const updatedPerPage = query[RHSM_API_QUERY_SET_TYPES.LIMIT] || perPageDefault;
    const updatedOffset = query[RHSM_API_QUERY_SET_TYPES.OFFSET];
    const isLastPage = paginationHelpers.isLastPage(updatedOffset, updatedPerPage, itemCount);

    // Set an updated key to force refresh minHeight
    const minHeightContentRefreshKey =
      (fulfilled === true && itemCount < updatedPerPage && `bodyMinHeight-${updatedPerPage}-resize`) ||
      (fulfilled === true && isLastPage && `bodyMinHeight-${updatedPerPage}-resize`) ||
      (error === true && `bodyMinHeight-${updatedPerPage}-resize`) ||
      `bodyMinHeight-${updatedPerPage}`;

    return (
      <Card className="curiosity-inventory-card">
        <MinHeight key="headerMinHeight" updateOnContent>
          <CardHeader className={(error && 'hidden') || ''} aria-hidden={error || false}>
            <CardHeaderMain>
              <ToolbarFieldDisplayName viewId={viewId} />
            </CardHeaderMain>
            <CardActions className={(!itemCount && 'transparent') || ''} aria-hidden={!itemCount || false}>
              <Pagination
                isCompact
                isDisabled={pending || error}
                itemCount={itemCount}
                offset={updatedOffset}
                onPage={this.onPage}
                onPerPage={this.onPage}
                perPage={updatedPerPage}
              />
            </CardActions>
          </CardHeader>
        </MinHeight>
        <MinHeight key={minHeightContentRefreshKey} updateOnContent>
          <CardBody>
            <div className={(error && 'blur') || (pending && 'fadein') || ''}>
              {pending && (
                <Loader
                  variant="table"
                  tableProps={{
                    className: 'curiosity-inventory-list',
                    colCount: filterInventoryData?.length || (listData?.[0] && Object.keys(listData[0]).length) || 1,
                    colWidth:
                      (filterInventoryData?.length && filterInventoryData.map(({ cellWidth }) => cellWidth)) || [],
                    rowCount: listData?.length || updatedPerPage,
                    variant: TableVariant.compact
                  }}
                />
              )}
              {!pending && this.renderTable()}
            </div>
          </CardBody>
        </MinHeight>
        <MinHeight key="footerMinHeight" updateOnContent>
          <CardFooter
            className={(error && 'hidden') || (!itemCount && 'transparent') || ''}
            aria-hidden={error || !itemCount || false}
          >
            <TableToolbar isFooter>
              <Pagination
                dropDirection="up"
                isDisabled={pending || error}
                itemCount={itemCount}
                offset={updatedOffset}
                onPage={this.onPage}
                onPerPage={this.onPage}
                perPage={updatedPerPage}
              />
            </TableToolbar>
          </CardFooter>
        </MinHeight>
      </Card>
    );
  }
}

/**
 * Prop types.
 *
 * @type {{settings: object, data: Array, productId: string, session: object, pending: boolean, query: object,
 *     fulfilled: boolean, error: boolean, getInstancesInventory: Function, viewId: string, t: Function,
 *     filterInventoryData: Array, meta: object, filterGuestsData: Array, perPageDefault: number,
 *     isDisabled: boolean}}
 */
InventoryList.propTypes = {
  data: PropTypes.array,
  error: PropTypes.bool,
  fulfilled: PropTypes.bool,
  filterGuestsData: PropTypes.array,
  filterInventoryData: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      header: PropTypes.oneOfType([
        PropTypes.shape({
          title: PropTypes.node.isRequired
        }),
        PropTypes.func,
        PropTypes.node
      ]),
      cell: PropTypes.oneOfType([
        PropTypes.shape({
          title: PropTypes.node.isRequired
        }),
        PropTypes.func,
        PropTypes.node
      ])
    }).isRequired
  ),
  getInstancesInventory: PropTypes.func,
  isDisabled: PropTypes.bool,
  meta: PropTypes.shape({ count: PropTypes.number }),
  pending: PropTypes.bool,
  perPageDefault: PropTypes.number,
  productId: PropTypes.string.isRequired,
  query: PropTypes.object.isRequired,
  session: PropTypes.object,
  settings: PropTypes.shape({
    hasGuests: PropTypes.func
  }),
  t: PropTypes.func,
  viewId: PropTypes.string
};

/**
 * Default props.
 *
 * @type {{settings: object, data: Array, session: object, pending: boolean, fulfilled: boolean, error: boolean,
 *     getInstancesInventory: Function, viewId: string, t: Function, filterInventoryData: Array, meta: object,
 *     filterGuestsData: Array, perPageDefault: number, isDisabled: boolean}}
 */
InventoryList.defaultProps = {
  data: [],
  error: false,
  fulfilled: false,
  filterGuestsData: [],
  filterInventoryData: [],
  getInstancesInventory: helpers.noop,
  isDisabled: helpers.UI_DISABLED_TABLE_INSTANCES,
  meta: {
    count: 0
  },
  pending: false,
  perPageDefault: 10,
  session: {},
  settings: {},
  t: translate,
  viewId: 'inventoryInstancesList'
};

/**
 * Apply actions to props.
 *
 * @param {Function} dispatch
 * @returns {object}
 */
const mapDispatchToProps = dispatch => ({
  getInstancesInventory: (id, query) => dispatch(reduxActions.rhsm.getInstancesInventory(id, query))
});

/**
 * Create a selector from applied state, props.
 *
 * @type {Function}
 */
const makeMapStateToProps = reduxSelectors.instancesList.makeInstancesList();

const ConnectedInventoryList = connect(makeMapStateToProps, mapDispatchToProps)(InventoryList);

export { ConnectedInventoryList as default, ConnectedInventoryList, InventoryList };
