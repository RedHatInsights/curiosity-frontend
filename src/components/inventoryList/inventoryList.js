import React from 'react';
import PropTypes from 'prop-types';
import _isEqual from 'lodash/isEqual';
import { TableVariant } from '@patternfly/react-table';
import { Card, CardActions, CardBody, CardFooter, CardHeader, CardTitle, Title } from '@patternfly/react-core';
import { helpers } from '../../common';
import { connect, reduxActions, reduxSelectors } from '../../redux';
import Table from '../table/table';
import { Loader } from '../loader/loader';
import { inventoryListHelpers } from './inventoryListHelpers';
import Pagination from '../pagination/pagination';
import { rhsmApiTypes } from '../../types';

class InventoryList extends React.Component {
  componentDidMount() {
    this.onUpdateInventoryData();
  }

  componentDidUpdate(prevProps) {
    const { productId, query } = this.props;

    if (productId !== prevProps.productId || !_isEqual(query, prevProps.query)) {
      this.onUpdateInventoryData();
    }
  }

  onUpdateInventoryData = () => {
    const { getHostsInventory, isDisabled, productId, query } = this.props;

    if (!isDisabled && productId) {
      getHostsInventory(productId, query);
    }
  };

  renderTable() {
    const { filterInventoryData, listData } = this.props;
    let updatedColumnHeaders = [];

    const updatedRows = listData.map(({ ...cellData }) => {
      const { columnHeaders, cells } = inventoryListHelpers.parseRowCellsListData({
        filters: filterInventoryData,
        cellData
      });

      updatedColumnHeaders = columnHeaders;

      return {
        cells
      };
    });

    return (
      <Table
        borders
        variant={TableVariant.compact}
        className="curiosity-inventory-list"
        columnHeaders={updatedColumnHeaders}
        rows={updatedRows}
      />
    );
  }

  render() {
    const {
      cardTitle,
      error,
      filterInventoryData,
      isDisabled,
      itemCount,
      listData,
      pending,
      perPageDefault,
      productId,
      query,
      viewId
    } = this.props;

    if (isDisabled) {
      return null;
    }

    const updatedPerPage = query?.[rhsmApiTypes.RHSM_API_QUERY_LIMIT] || perPageDefault;
    const loaderPerPage = updatedPerPage / 2;

    return (
      <Card className="curiosity-inventory-card">
        <CardHeader>
          <CardTitle>
            <Title headingLevel="h2" size="lg">
              {cardTitle}
            </Title>
          </CardTitle>
          <CardActions className={(error && 'blur') || ''}>
            <Pagination
              isCompact
              isDisabled={pending || error}
              itemCount={itemCount}
              productId={productId}
              viewId={viewId}
              itemsPerPageDefault={updatedPerPage}
            />
          </CardActions>
        </CardHeader>
        <CardBody>
          <div className={(error && 'blur') || 'fadein'}>
            {pending && (
              <Loader
                variant="table"
                tableProps={{
                  className: 'curiosity-inventory-list',
                  colCount: filterInventoryData?.length || (listData?.[0] && Object.keys(listData[0]).length) || 1,
                  rowCount: listData?.length || loaderPerPage,
                  variant: TableVariant.compact
                }}
              />
            )}
            {!pending && this.renderTable()}
          </div>
        </CardBody>
        <CardFooter className={(error && 'blur') || ''}>
          <Pagination
            isDisabled={pending || error}
            itemCount={itemCount}
            productId={productId}
            viewId={viewId}
            perPageDefault={updatedPerPage}
            dropDirection="up"
          />
        </CardFooter>
      </Card>
    );
  }
}

InventoryList.propTypes = {
  error: PropTypes.bool,
  cardTitle: PropTypes.string,
  filterInventoryData: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      header: PropTypes.oneOfType([
        PropTypes.shape({
          title: PropTypes.string
        }),
        PropTypes.func
      ]),
      cell: PropTypes.oneOfType([
        PropTypes.shape({
          title: PropTypes.string
        }),
        PropTypes.func
      ])
    }).isRequired
  ),
  getHostsInventory: PropTypes.func,
  isDisabled: PropTypes.bool,
  itemCount: PropTypes.number,
  listData: PropTypes.array,
  pending: PropTypes.bool,
  productId: PropTypes.string.isRequired,
  perPageDefault: PropTypes.number,
  query: PropTypes.object.isRequired,
  viewId: PropTypes.string
};

InventoryList.defaultProps = {
  error: false,
  cardTitle: null,
  filterInventoryData: [],
  getHostsInventory: helpers.noop,
  isDisabled: helpers.UI_DISABLED_TABLE,
  itemCount: 0,
  listData: [],
  pending: false,
  perPageDefault: 10,
  viewId: 'inventoryList'
};

/**
 * Apply actions to props.
 *
 * @param {Function} dispatch
 * @returns {object}
 */
const mapDispatchToProps = dispatch => ({
  getHostsInventory: (id, query) => dispatch(reduxActions.rhsm.getHostsInventory(id, query))
});

/**
 * Create a selector from applied state, props.
 *
 * @type {Function}
 */
const makeMapStateToProps = reduxSelectors.inventoryList.makeInventoryList();

const ConnectedInventoryList = connect(makeMapStateToProps, mapDispatchToProps)(InventoryList);

export { ConnectedInventoryList as default, ConnectedInventoryList, InventoryList };
