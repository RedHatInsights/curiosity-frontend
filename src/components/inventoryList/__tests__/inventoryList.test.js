import React from 'react';
import { shallow } from 'enzyme';
import { InventoryList } from '../inventoryList';
import { rhsmApiTypes } from '../../../types';

describe('InventoryList Component', () => {
  it('should render a non-connected component', () => {
    const props = {
      query: {
        [rhsmApiTypes.RHSM_API_QUERY_LIMIT]: 10,
        [rhsmApiTypes.RHSM_API_QUERY_OFFSET]: 0
      },
      productId: 'lorem'
    };

    const component = shallow(<InventoryList {...props} />);
    expect(component).toMatchSnapshot('non-connected');
  });

  it('should handle variations in data', () => {
    const props = {
      query: {
        [rhsmApiTypes.RHSM_API_QUERY_LIMIT]: 10,
        [rhsmApiTypes.RHSM_API_QUERY_OFFSET]: 0
      },
      productId: 'lorem',
      listData: [
        { lorem: 'ipsum', dolor: 'sit' },
        { lorem: 'sit', dolor: 'amet' }
      ]
    };

    const component = shallow(<InventoryList {...props} />);
    expect(component).toMatchSnapshot('variable data');

    component.setProps({
      filterInventoryData: [{ id: 'lorem' }]
    });

    expect(component).toMatchSnapshot('filtered data');
  });
});
