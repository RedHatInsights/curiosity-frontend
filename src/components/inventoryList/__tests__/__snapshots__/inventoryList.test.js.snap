// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryList Component should handle expandable guests data: number of guests 1`] = `
<Table
  ariaLabel={null}
  borders={true}
  className="curiosity-inventory-list"
  columnHeaders={
    Array [
      "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
      "t(curiosity-inventory.header, {\\"context\\":\\"numberOfGuests\\"})",
    ]
  }
  isHeader={true}
  rows={
    Array [
      Object {
        "cells": Array [
          "sit",
          "amet",
          1,
        ],
        "expandedContent": undefined,
      },
    ]
  }
  summary={null}
  t={[Function]}
  variant="compact"
/>
`;

exports[`InventoryList Component should handle expandable guests data: number of guests, and id 1`] = `
<Table
  ariaLabel={null}
  borders={true}
  className="curiosity-inventory-list"
  columnHeaders={
    Array [
      "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
      "t(curiosity-inventory.header, {\\"context\\":\\"numberOfGuests\\"})",
      "t(curiosity-inventory.header, {\\"context\\":\\"subscriptionManagerId\\"})",
    ]
  }
  isHeader={true}
  rows={
    Array [
      Object {
        "cells": Array [
          "sit",
          "amet",
          1,
          "loremIpsum",
        ],
        "expandedContent": <UNDEFINED
          filterGuestsData={Array []}
          id="loremIpsum"
          numberOfGuests={1}
          query={
            Object {
              "limit": 10,
              "offset": 0,
            }
          }
        />,
      },
    ]
  }
  summary={null}
  t={[Function]}
  variant="compact"
/>
`;

exports[`InventoryList Component should handle expandable guests data: number of guests, id, and NO expandable guests display 1`] = `
<Table
  ariaLabel={null}
  borders={true}
  className="curiosity-inventory-list"
  columnHeaders={
    Array [
      "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
      "t(curiosity-inventory.header, {\\"context\\":\\"numberOfGuests\\"})",
      "t(curiosity-inventory.header, {\\"context\\":\\"subscriptionManagerId\\"})",
    ]
  }
  isHeader={true}
  rows={
    Array [
      Object {
        "cells": Array [
          "sit",
          "amet",
          2,
          "loremIpsum",
        ],
        "expandedContent": undefined,
      },
    ]
  }
  summary={null}
  t={[Function]}
  variant="compact"
/>
`;

exports[`InventoryList Component should handle updating sorting through redux state: dispatch filter 1`] = `
Array [
  Array [
    Array [
      Object {
        "dir": "asc",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_dir",
        "viewId": "lorem",
      },
      Object {
        "sort": "sockets",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_sort",
        "viewId": "lorem",
      },
    ],
  ],
  Array [
    Array [
      Object {
        "dir": "desc",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_dir",
        "viewId": "lorem",
      },
      Object {
        "sort": "sockets",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_sort",
        "viewId": "lorem",
      },
    ],
  ],
  Array [
    Array [
      Object {
        "dir": "asc",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_dir",
        "viewId": "lorem",
      },
      Object {
        "sort": "sockets",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_sort",
        "viewId": "lorem",
      },
    ],
  ],
]
`;

exports[`InventoryList Component should handle variations in data: filtered data 1`] = `
<Card
  className="curiosity-inventory-card"
>
  <MinHeight
    key="headerMinHeight"
    minHeight={0}
    updateOnContent={true}
    updateOnResize={true}
  >
    <CardHeader>
      <CardActions
        className=""
      >
        <Pagination
          dropDirection="down"
          isCompact={true}
          isDisabled={false}
          itemCount={0}
          limitType="SET_QUERY_RHSM_HOSTS_INVENTORY_limit"
          offsetDefault={0}
          offsetType="SET_QUERY_RHSM_HOSTS_INVENTORY_offset"
          perPageDefault={10}
          productId="lorem"
          query={
            Object {
              "limit": 10,
              "offset": 0,
            }
          }
          variant={null}
          viewId="inventoryList"
        />
      </CardActions>
    </CardHeader>
  </MinHeight>
  <MinHeight
    key="bodyMinHeight-10"
    minHeight={0}
    updateOnContent={true}
    updateOnResize={true}
  >
    <CardBody>
      <div
        className="fadein"
      >
        <Table
          ariaLabel={null}
          borders={true}
          className="curiosity-inventory-list"
          columnHeaders={
            Array [
              Object {
                "title": "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
                "transforms": Array [],
              },
            ]
          }
          isHeader={true}
          rows={
            Array [
              Object {
                "cells": Array [
                  Object {
                    "title": "ipsum",
                  },
                ],
                "expandedContent": undefined,
              },
              Object {
                "cells": Array [
                  Object {
                    "title": "sit",
                  },
                ],
                "expandedContent": undefined,
              },
            ]
          }
          summary={null}
          t={[Function]}
          variant="compact"
        />
      </div>
    </CardBody>
  </MinHeight>
  <MinHeight
    key="footerMinHeight"
    minHeight={0}
    updateOnContent={true}
    updateOnResize={true}
  >
    <CardFooter
      className=""
    >
      <TableToolbar
        isFooter={true}
      >
        <Pagination
          dropDirection="up"
          isCompact={false}
          isDisabled={false}
          itemCount={0}
          limitType="SET_QUERY_RHSM_HOSTS_INVENTORY_limit"
          offsetDefault={0}
          offsetType="SET_QUERY_RHSM_HOSTS_INVENTORY_offset"
          perPageDefault={10}
          productId="lorem"
          query={
            Object {
              "limit": 10,
              "offset": 0,
            }
          }
          variant={null}
          viewId="inventoryList"
        />
      </TableToolbar>
    </CardFooter>
  </MinHeight>
</Card>
`;

exports[`InventoryList Component should handle variations in data: variable data 1`] = `
<Card
  className="curiosity-inventory-card"
>
  <MinHeight
    key="headerMinHeight"
    minHeight={0}
    updateOnContent={true}
    updateOnResize={true}
  >
    <CardHeader>
      <CardActions
        className=""
      >
        <Pagination
          dropDirection="down"
          isCompact={true}
          isDisabled={false}
          itemCount={0}
          limitType="SET_QUERY_RHSM_HOSTS_INVENTORY_limit"
          offsetDefault={0}
          offsetType="SET_QUERY_RHSM_HOSTS_INVENTORY_offset"
          perPageDefault={10}
          productId="lorem"
          query={
            Object {
              "limit": 10,
              "offset": 0,
            }
          }
          variant={null}
          viewId="inventoryList"
        />
      </CardActions>
    </CardHeader>
  </MinHeight>
  <MinHeight
    key="bodyMinHeight-10"
    minHeight={0}
    updateOnContent={true}
    updateOnResize={true}
  >
    <CardBody>
      <div
        className="fadein"
      >
        <Table
          ariaLabel={null}
          borders={true}
          className="curiosity-inventory-list"
          columnHeaders={
            Array [
              "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
              "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
            ]
          }
          isHeader={true}
          rows={
            Array [
              Object {
                "cells": Array [
                  "ipsum",
                  "sit",
                ],
                "expandedContent": undefined,
              },
              Object {
                "cells": Array [
                  "sit",
                  "amet",
                ],
                "expandedContent": undefined,
              },
            ]
          }
          summary={null}
          t={[Function]}
          variant="compact"
        />
      </div>
    </CardBody>
  </MinHeight>
  <MinHeight
    key="footerMinHeight"
    minHeight={0}
    updateOnContent={true}
    updateOnResize={true}
  >
    <CardFooter
      className=""
    >
      <TableToolbar
        isFooter={true}
      >
        <Pagination
          dropDirection="up"
          isCompact={false}
          isDisabled={false}
          itemCount={0}
          limitType="SET_QUERY_RHSM_HOSTS_INVENTORY_limit"
          offsetDefault={0}
          offsetType="SET_QUERY_RHSM_HOSTS_INVENTORY_offset"
          perPageDefault={10}
          productId="lorem"
          query={
            Object {
              "limit": 10,
              "offset": 0,
            }
          }
          variant={null}
          viewId="inventoryList"
        />
      </TableToolbar>
    </CardFooter>
  </MinHeight>
</Card>
`;

exports[`InventoryList Component should render a non-connected component: non-connected 1`] = `
<Card
  className="curiosity-inventory-card"
>
  <MinHeight
    key="headerMinHeight"
    minHeight={0}
    updateOnContent={true}
    updateOnResize={true}
  >
    <CardHeader>
      <CardActions
        className=""
      >
        <Pagination
          dropDirection="down"
          isCompact={true}
          isDisabled={false}
          itemCount={0}
          limitType="SET_QUERY_RHSM_HOSTS_INVENTORY_limit"
          offsetDefault={0}
          offsetType="SET_QUERY_RHSM_HOSTS_INVENTORY_offset"
          perPageDefault={10}
          productId="lorem"
          query={
            Object {
              "limit": 10,
              "offset": 0,
            }
          }
          variant={null}
          viewId="inventoryList"
        />
      </CardActions>
    </CardHeader>
  </MinHeight>
  <MinHeight
    key="bodyMinHeight-10"
    minHeight={0}
    updateOnContent={true}
    updateOnResize={true}
  >
    <CardBody>
      <div
        className="fadein"
      >
        <Table
          ariaLabel={null}
          borders={true}
          className="curiosity-inventory-list"
          columnHeaders={Array []}
          isHeader={true}
          rows={Array []}
          summary={null}
          t={[Function]}
          variant="compact"
        />
      </div>
    </CardBody>
  </MinHeight>
  <MinHeight
    key="footerMinHeight"
    minHeight={0}
    updateOnContent={true}
    updateOnResize={true}
  >
    <CardFooter
      className=""
    >
      <TableToolbar
        isFooter={true}
      >
        <Pagination
          dropDirection="up"
          isCompact={false}
          isDisabled={false}
          itemCount={0}
          limitType="SET_QUERY_RHSM_HOSTS_INVENTORY_limit"
          offsetDefault={0}
          offsetType="SET_QUERY_RHSM_HOSTS_INVENTORY_offset"
          perPageDefault={10}
          productId="lorem"
          query={
            Object {
              "limit": 10,
              "offset": 0,
            }
          }
          variant={null}
          viewId="inventoryList"
        />
      </TableToolbar>
    </CardFooter>
  </MinHeight>
</Card>
`;

exports[`InventoryList Component should return an empty render when disabled: disabled component 1`] = `
<Card
  className="curiosity-inventory-card__disabled"
>
  <CardBody>
    <Bullseye>
      t(curiosity-inventory.tab, {"context":"disabled"})
    </Bullseye>
  </CardBody>
</Card>
`;
